truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/parser/NodeParser.java

```diff
@@ -463,6 +463,11 @@ private NodeData parseNodeData(TypeElement templateType, List<TypeElement> typeH
                 } else if (names.contains(name)) {
                     field.addError(ElementUtils.getAnnotationValue(mirror, "name"), "Duplicate field name '%s'.", name);
                 }
+
+                if (type == null) {
+                    throw new AssertionError("The type specified in the NodeField annotation cannot be found. A possible cause could be a missing import statement for the type.");
```

Comment:
Also, I suggest introducing a consistent style for error messages with respect to format, style, capitalization, punctuation and grammar. For example,

"%s: type used in NodeField annotation not found (missing import?)"

Separating the variables contained in messages would also allow for i18n.